pylint/extensions/eq_without_hash.py

```diff
@@ -0,0 +1,39 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+
+"""This is the remnant of the python3 checker. It was removed because
+the transition from python 2 to python3 is behind us, but some checks
+are still useful in python3 after all.
+See https://github.com/PyCQA/pylint/issues/5025
+"""
+
+from astroid import nodes
+
+from pylint import checkers, interfaces
+from pylint.checkers import utils
+from pylint.lint import PyLinter
+
+
+class EqWithoutHash(checkers.BaseChecker):
+
+    __implements__ = interfaces.IAstroidChecker
+    name = "eq-without-hash"
+
+    msgs = {
+        "W1641": (
+            "Implementing __eq__ without also implementing __hash__",
+            "eq-without-hash",
+            "Used when a class implements __eq__ but not __hash__. Objects get "
+            "None as their default __hash__ implementation if they also implement __eq__.",
+        ),
+    }
+
+    @utils.check_messages("eq-without-hash")
+    def visit_classdef(self, node: nodes.ClassDef) -> None:
+        locals_and_methods = set(node.locals).union(x.name for x in node.mymethods())
+        if "__eq__" in locals_and_methods and "__hash__" not in locals_and_methods:
+            self.add_message("eq-without-hash", node=node)
```

Comment:
Do we want to add a confidence while we're at it?
Does `mymethods()` use inference? If not I think this should be `HIGH`.

Reply:
It's only yielding FunctionDef:
```python
    def mymethods(self):
        """Iterate over all of the method defined in this class only.

        :returns: The methods defined on the class.
        :rtype: iterable(FunctionDef)
        """
        for member in self.values():
            if isinstance(member, FunctionDef):
                yield member
```
```suggestion
            self.add_message("eq-without-hash", node=node, confidence=interfaces.HIGH)
```